include/vara/Feature/Error.h

```diff
@@ -0,0 +1,79 @@
+#ifndef VARA_FEATURE_ERROR_H
+#define VARA_FEATURE_ERROR_H
+
+#include "llvm/Support/raw_ostream.h"
+
+#include <cassert>
+#include <iostream>
+#include <type_traits>
+#include <variant>
+
+namespace vara::feature {
+
+enum ErrorCode {
```

Comment:
```suggestion
enum FTErrorCode {
```
We should make it a bit clearer to what the ErrorCode actually belongs. So maybe `FTErrorCode` or `FeatureTransactionErrorCode` or `FeatureErrorCode`